packages/node/src/localEvaluation.ts

```diff
@@ -0,0 +1,188 @@
+ * @category Core Usage
+ */
+export class LocalEvaluationClient {
+  private readonly apiKey: string;
+  private readonly httpClient: HttpClient;
+  private readonly config: LocalEvaluationConfig;
+  private readonly flagConfigCache: FlagConfigCache;
+
+  private flagConfigPoller: NodeJS.Timeout;
+  private flagConfigPromise: Promise<void>;
+
+  /**
+   * Construct a local evaluation client.
+   *
+   * Call {@link start} to fetch the flag configs and begin polling for updates,
+   * and {@link evaluate} to evaluate a user for a set of flags.
+   *
+   * @param apiKey The environment's api key.
+   * @param config The {@link LocalEvaluationConfig}.
+   * @param flagConfigCache The cache used to access flag configs to evaluate.
+   */
+  public constructor(
+    apiKey: string,
+    config: LocalEvaluationConfig,
+    flagConfigCache: FlagConfigCache,
+  ) {
+    this.apiKey = apiKey;
+    this.config = { ...LocalEvaluationDefaults, ...config };
+    this.httpClient = FetchHttpClient;
+    this.flagConfigCache = flagConfigCache;
+  }
+
+  /**
+   * Locally evaluates flag variants for a user.
+   *
+   * This function will only evaluate flags for the keys specified in the
+   * {@link flags} argument. If the {@link flags} argument is missing, all flags
+   * in the {@link FlagConfigCache} will be evaluated.
+   *
+   * @param user The user to evaluate
+   * @param flags The flags to evaluate with the user. If empty, all flags from
+   * the flag cache are evaluated.
+   * @returns The evaluated variants
+   */
+  public async evaluate(
+    user: ExperimentUser,
+    flags?: string[],
+  ): Promise<Variants> {
+    // Evaluate the flag configs and user.
+    const flagConfigs = await this.getFlagConfigs(flags);
+    this.debug('[Experiment] evaluate - user:', user, 'flags:', flags);
+    const resultsString = evaluation.evaluate(
+      JSON.stringify(flagConfigs),
+      JSON.stringify(user),
+    );
+    this.debug('[Experiment] evaluate - result:', resultsString);
+    // Parse variant results
+    const results: EvaluationResult = JSON.parse(resultsString);
+    const variants: Variants = {};
+    Object.keys(results).forEach((key) => {
+      variants[key] = {
+        value: results[key].variant.key,
+        payload: results[key].variant.payload,
+      };
+    });
+    return variants;
+  }
+
+  /**
+   * Fetch initial flag configurations and start polling for updates.
+   *
+   * You must call this function to begin polling for flag config updates.
+   * The promise returned by this function is resolved when the initial call
+   * to fetch the flag configuration completes.
+   *
+   * Calling this function while the poller is already running will stop and
+   * restart the poller.
+   */
+  public async start(): Promise<void> {
+    this.stop();
+    this.debug('[Experiment] poller - start');
+    this.flagConfigPoller = setInterval(async () => {
+      await this.updateFlagConfigs();
+    }, this.config.flagConfigPollingIntervalMillis);
```

Comment:
The poll frequency is high enough some convoluted chain of network timeouts could cause requests to start queuing up.

I would recommend triggering a `setTimeout` from the end of the previous poll loop to prevent this.

Reply:
I had a bug here regarding the backoff logic for polling requests. That has been resolved, but I am still not sure I follow the difference between `setInterval` and calling `setTimeout` within a `setTimeout`?

Reply:
`setInterval` is fire and forget. It won't care if the side effects from the previous callback are still running, it'll fire again anyway.

If you want to guarantee that you only ever have one callback running at a time then you need to use `setTimeout`.